var/spack/repos/builtin/packages/tcl/package.py

```diff
@@ -49,12 +49,19 @@ def url_for_version(self, version):
         base_url = 'http://prdownloads.sourceforge.net/tcl'
         return '{0}/tcl{1}-src.tar.gz'.format(base_url, version)
 
-    def setup_environment(self, spack_env, run_env):
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
```

Comment:
So I added these `setup_environment` methods to tcl and tk back in #1226, but that was back when I didn't have a good understanding of how they worked. Shouldn't it be `setup_dependent_environment`? Or both? `run_env` or `spack_env`?